x-pack/plugins/fleet/server/routes/uninstall_token/handlers.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { TypeOf } from '@kbn/config-schema';
+
+import type { PolicyUninstallTokenMap } from '../../../common/types/models/uninstall_token';
+
+import { appContextService } from '../../services';
+import type { UninstallTokenServiceInterface } from '../../services/security/uninstall_token_service';
+import type { FleetRequestHandler } from '../../types';
+import type { GetUninstallTokensResponse } from '../../../common/types/rest_spec/uninstall_token';
+import type { GetUninstallTokensRequestSchema } from '../../types/rest_spec/uninstall_token';
+
+export const getUninstallTokensHandler: FleetRequestHandler<
+  unknown,
+  TypeOf<typeof GetUninstallTokensRequestSchema.query>
+> = async (context, request, response) => {
+  const { agentTamperProtectionEnabled } = appContextService.getExperimentalFeatures();
+  if (!agentTamperProtectionEnabled) {
+    return response.customError({
+      statusCode: 404,
+      body: { message: 'Not Found' },
+    });
+  }
+
+  const uninstallTokenService = appContextService.getUninstallTokenService();
+  if (!uninstallTokenService) {
+    return response.customError({
+      statusCode: 500,
+      body: { message: 'Uninstall Token Service is unavailable.' },
+    });
+  }
+
+  try {
+    const uninstallTokens = await readUninstallTokens(
+      uninstallTokenService,
+      request.query.policyId
```

Comment:
will this work if the policyId query param is a partial policyId?

Reply:
nope - I'll add that functionality. somehow I missed the clarification on this topic, so thanks for bringing this up!

Reply:
added here 84936be